core/src/main/java/org/keycloak/representations/account/UserRepresentation.java

```diff
@@ -106,4 +107,28 @@ public String firstAttribute(String key) {
         return this.attributes == null ? null : this.attributes.containsKey(key) ? this.attributes.get(key).get(0) : null;
     }
 
+    public Map<String, List<String>> toAttributes() {
+        Map<String, List<String>> attrs = new HashMap<>();
+
+        if (getAttributes() != null) attrs.putAll(getAttributes());
+
+        if (getUsername() != null)
+            attrs.put("username", Collections.singletonList(getUsername()));
```

Comment:
Can be constants for these four attributes (username, email, firstName, lastName) defined as constants at one place, ideally somewhere in UserProfileSPI. Mainly username and email will be still a bit special with some special handling in UserProfileSPI code

Reply:
I have two considerations here:

* We should move the common properties that today are in `UserModel` to the user profile. They are email, first/last names, username, etc. 
* We could allow creating a `UserProfile` based on the user representations. It would be another method to `UserProfileProvider`.

The problem with #1 is that SPI is still private and we need access from public spis.

The thing about #2 is that we would need to unify user representation, possibly using the type from the admin rest. That would require us to remove the specific representation currently used by the Account API.

Another thing is that we can also support in user profile the generation of representations. Where the representation could also change depending on the context/config.